### PR TITLE
build: remove some warnings from the Maven build

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: build/build-parent/pom.xml [hunk 1]
```diff
@@ -329,17 +334,6 @@
...
-            <analysisConfigurationFiles>
-              <configurationFile>
-                <path>src/build/revapi-config.json</path>
-                <roots>
-                  <root>filters</root>
-                  <root>ignores</root>
-                </roots>
-              </configurationFile>
-            </analysisConfigurationFiles>
-            <failSeverity>potentiallyBreaking</failSeverity>
```
Comment (rsynek) [inline build/build-parent/pom.xml:342]: Is this configuration no longer relevant?

File: quarkus-integration/quarkus-benchmark/runtime/src/main/java/ai/timefold/solver/benchmark/quarkus/TimefoldBenchmarkRecorder.java [hunk 2]
```diff
@@ -178,7 +178,6 @@ private boolean isScoreCalculationDefined(SolverConfig solverConfig) {
         }
         return scoreDirectorFactoryConfig.getEasyScoreCalculatorClass() != null ||
                 scoreDirectorFactoryConfig.getIncrementalScoreCalculatorClass() != null ||
-                scoreDirectorFactoryConfig.getScoreDrlList() != null ||
```
Comment (rsynek) [inline quarkus-integration/quarkus-benchmark/runtime/src/main/java/ai/timefold/solver/benchmark/quarkus/TimefoldBenchmarkRecorder.java:181]: Good catch

File: test/pom.xml [hunk 3]
```diff
@@ -72,6 +72,12 @@
       <plugin>
         <groupId>org.revapi</groupId>
         <artifactId>revapi-maven-plugin</artifactId>
+        <configuration>
+          <analysisConfigurationFiles>
+            <file>${project.basedir}/src/build/revapi-filter.json</file>
+            <file>${project.basedir}/src/build/revapi-differences.json</file>
+          </analysisConfigurationFiles>
+        </configuration>
```
Comment (rsynek) [inline test/pom.xml:80]: Why not define this config in the build parent pom.xml?

**Review by rsynek — APPROVED** [review]
LGTM; consider moving the repeating revapi configuration in the build parent pom.xml.

File: build/build-parent/pom.xml (hunk 1 above)
Comment (triceo) [reply build/build-parent/pom.xml:342]: The configuration, turns out, is the default.

**Pull request merged**